xwiki-platform-core/xwiki-platform-edit/xwiki-platform-edit-ui/src/main/resources/XWiki/InplaceEditing.xml

```diff
@@ -873,45 +879,33 @@ define('editInPlace', [
       return this._getActionButtons().find(selector)[0];
     },
     serialize: function() {
-      var extra = this._getActionButtons().find(':input').serializeArray().reduce(function(extra, entry) {
-        var value = extra[entry.name] || [];
-        value.push(entry.value);
-        extra[entry.name] = value;
-        return extra;
-      }, {});
+      var formData = new FormData();
+      addToFormData(formData, this._getActionButtons().find(':input'));
       // retrieve all input fields listing the temporary uploaded files.
-      var uploadedFiles = $('#xwikicontent').nextAll('input[name="uploadedFiles"]').serializeArray().reduce(function(extra, entry) {
-        var value = extra[entry.name] || [];
-        value.push(entry.value);
-        extra[entry.name] = value;
-        return extra;
-      }, {});
       var xwikiDocument = this._getActionButtons().data('xwikiDocument');
-      var formData = {
-        title: xwikiDocument.rawTitle,
-        language: xwikiDocument.getRealLocale(),
-        isNew: xwikiDocument.isNew
-      };
-      if (xwikiDocument.content != xwikiDocument.originalDocument.content) {
+      formData.append('title', xwikiDocument.rawTitle);
+      formData.append('language', xwikiDocument.getRealLocale());
+      formData.append('isNew', xwikiDocument.isNew);
```

Comment:
The order in the original code is very important. The form data set explicitly by the in-place editor **must overwrite** whatever comes from action buttons input (which the editor doesn't control), otherwise the editor can't be sure that the values it adds are the ones that will be used on the server. This is not hypothetical, the editor has to overwrite stuff like ``isNew``, ``previousVersion`` and ``editingVersionDate``, see https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-14.9/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/editactions.vm#L43-L50 .

```suggestion
      formData.set('title', xwikiDocument.rawTitle);
      formData.set('language', xwikiDocument.getRealLocale());
      formData.set('isNew', xwikiDocument.isNew);
```